src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/utils/StringValidators.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.companieshouse.overseasentitiesapi.validation.utils;
+
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.util.regex.Pattern;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators.setErrorMsgToLocation;
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators.validateNotNull;
+
+public final class StringValidators {
+
+    private StringValidators() {}
+
+    public static boolean validateStringNotBlank(String toTest, String location, Errors errs, String loggingContext) {
+        return validateNotNull(toTest, location, errs, loggingContext) && validateStringNotEmpty(toTest, location, errs, loggingContext);
+    }
+
+    public static boolean validateMaxLength(String toTest, Integer maxLength, String location, Errors errs, String loggingContext) {
+        if (toTest.length() > maxLength) {
+            setErrorMsgToLocation(errs, location,
+                    location + ValidationMessages.MAX_LENGTH_EXCEEDED_ERROR_MESSAGE.replace("%s", maxLength.toString()));
+            ApiLogger.infoContext(loggingContext, "Invalid length for " + location);
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validateCharacters(String toTest, String location, Errors errs, String loggingContext) {
+        var regex = "^[-,.:; 0-9A-Z&@$£¥€'\"«»?!/\\\\()\\[\\]{}<>*=#%+ÀÁÂÃÄÅĀĂĄÆǼÇĆĈĊČÞĎÐÈÉÊËĒĔĖĘĚĜĞĠĢĤĦÌÍÎÏĨĪĬĮİĴĶĹĻĽĿŁÑŃŅŇŊÒÓÔÕÖØŌŎŐǾŒŔŖŘŚŜŞŠŢŤŦÙÚÛÜŨŪŬŮŰŲŴẀẂẄỲÝŶŸŹŻŽa-zſƒǺàáâãäåāăąæǽçćĉċčþďðèéêëēĕėęěĝģğġĥħìíîïĩīĭįĵķĺļľŀłñńņňŋòóôõöøōŏőǿœŕŗřśŝşšţťŧùúûüũūŭůűųŵẁẃẅỳýŷÿźżž]*$";
+
+        var pattern = Pattern.compile(regex);
+        var matcher = pattern.matcher(toTest);
+
+        if (!matcher.matches()) {
+            setErrorMsgToLocation(errs, location, location + ValidationMessages.INVALID_CHARACTERS_ERROR_MESSAGE);
+            ApiLogger.infoContext(loggingContext, "Invalid characters for " + location);
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validateEmail(String email, String location, Errors errs, String loggingContext) {
+        // The email regex is taken from OWASP https://owasp.org/www-community/OWASP_Validation_Regex_Repository
```

Comment:
Move this useful comment into a Javadoc comment on the method. When dealing with email addresses I like to get the 'address' part into the name as it's not the actual email we're validating. How about 'isValidEmailAddress' for the method name?

Reply:
Added a javadoc.